pytorch_lightning/lite/wrappers.py

```diff
@@ -95,12 +95,17 @@ def forward(self, *args: Any, **kwargs: Any) -> Any:
         }
         # TODO (@awaelchli): let the precision plugin handle the conversion
         to_type = precision_to_type[precision]
-        args, kwargs = apply_to_collection([args, kwargs], function=lambda t: t.to(to_type), dtype=Tensor)
+
+        def _convert_float_tensor(t: Tensor) -> Tensor:
+            return t.to(to_type) if torch.is_floating_point(t) else t
```

Comment:
noob question: If you provide both float32, and int32 and request precision=16 or "bf16": torch.bfloat16. Would this convert only the float32 -> float16 but leave the int32 unchanged ?

Reply:
Yes. This is why #10079 is important, an example:

```python
import torch

torch.set_default_dtype(torch.double)
x = torch.tensor(1)
print(x.dtype)  # int64
```

which is what you would expect.

Reply:
Also, this should probably differentiate between `AMP` 16, bf16 vs full 16, bf16 conversion

Reply:
That's great, but #10079 and Carlos's explanation is more about the creation of tensors if I understand correctly. Here we are concerned about the conversion going in and out of forward. Where the tensors are created is out of control of Lite (could be inside or outside run). If #10079 lands it can be done for Lite easily.